src/queue.rs

```diff
@@ -0,0 +1,98 @@
+use core::{
```

Comment:
Please move module `queue` as a sub-module inside the `task` module. All synchronization primitives in the HAL are currently inside `task` - either directly, or as sub-modules.